lambda/translcore.ml

```diff
@@ -731,7 +727,7 @@ and transl_apply ~scopes
         in
         let id_arg = Ident.create_local "param" in
         let body =
-          match build_apply handle ((Lvar id_arg, optional)::args') l with
+          match build_apply handle [Lvar id_arg, optional] l with
             Lfunction{kind = Curried; params = ids; return;
```

Comment:
I think that `build_apply` can now only return `Lfunction` with an empty list for `args`. That means that this branch and the next are no longer reachable.

Reply:
I think so. Your example needs to generate the equivalent of:
```ocaml
let z = foo in
fun ~x ->
  let f' = f ~x in
  fun ~y ->
  f' ~x ~y ~z
```
in case there are side-effects in `f` after the `x` is applied.

Reply:
Ah, you're right. I'll remove the unused cases. This code was building curried functions without checking the `max_arity` setting anyway.